bugbug/models/bugseverity.py

```diff
@@ -0,0 +1,163 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+
+import xgboost
+from imblearn.over_sampling import BorderlineSMOTE
+from sklearn.compose import ColumnTransformer
+from sklearn.feature_extraction import DictVectorizer
+from sklearn.pipeline import Pipeline
+from tabulate import tabulate
+
+from bugbug import bug_features, bugzilla, feature_cleanup, utils
+from bugbug.model import BugModel
+
+OPT_MSG_MISSING = (
+    "Optional dependencies are missing, install them with: pip install bugbug[nlp]\n"
+)
+
+try:
+    import nltk
+    from nltk.tokenize import word_tokenize
+except ImportError:
+    raise ImportError(OPT_MSG_MISSING)
+
+nltk.download("punkt")
+
+logger = logging.getLogger(__name__)
+
+SEVERITY_MAP = {
+    "S1": 1,
+    "critical": 1,
+    "blocker": 1,
+    "S2": 2,
+    "major": 2,
+    "S3": 3,
+    "normal": 3,
+    "S4": 4,
+    "minor": 4,
+    "trivial": 4,
+    "enhancement": 4,
+}
+
+# Keywords should be in lower case.
+SEVERITY_KEYWORDS = set(map(lambda x: x.lower(), SEVERITY_MAP.keys())) | {
+    "severity",
+}
```

Comment:
We should also try excluding bugs where severity is set since the beginning from the test set and add an option to include/exclude them in the training set (so we can compare the results when including/excluding them)